libass/ass_parse.c

```diff
@@ -1025,6 +1025,42 @@ unsigned get_next_char(ASS_Renderer *render_priv, char **str)
     return chr;
 }
 
+#ifdef CONFIG_LIBUNIBREAK
+utf32_t text_info_get_next_char_utf32(
+        TextInfo *text_info,
+        size_t len,
+        size_t *ip)
+{
+    GlyphInfo *cur;
+    unsigned ch;
+
+    assert(*ip <= len);
+    if (*ip == len)
+        return EOS;
+
+    cur = text_info->glyphs + *ip;
+    ch = cur->symbol;
+    *ip += 1;
+
+    return ch;
+}
+
+int is_valid_ub_lang(char *lang)
```

Comment:
Is this actually neccessary ?
If libunibreak doesn't recognize the language string it seeems to default to NULL anyway (as you are manually doing right now).  
https://github.com/adah1972/libunibreak/blob/3bf54a439e0224e37d84ec92ec652f1609b10e49/src/linebreak.c#L377

Should libass change the language tag to accept BCP 47 language tags with variants, you might want to reject all language codes whose base code (without variants) is longer than 2 letters to avoid false negatives.  
See: https://github.com/libass/libass/issues/404#issuecomment-648306753